### ISSUE TITLE
Elm.Input.checkbox is not accessible for screen readers

### ISSUE THREAD

**Issue by SamKacer** [mdgriffith/elm-ui#236]
## the problem

Every checkbox created using `Element.Input.checkbox` isn't presented nicely to screen readers. Note following was tested using NVDA and Windows Narrator on chrome and firefox. 

This is how a normal checkbox is presented by NVDA (narrator is same, but label and checkbox state in different order):
html: 
```
<label>
    classic input checkbox
	   <input type="checkbox" />
</label>
```

NVDA output:
"checkbox not checked classic input checkbox"

However a elm UI checkbox is presented like this:
NVDA output:
"classic input checkbox
blank"

narrator output:
"classic input checkbox
checkbox not checked"

Two issues. The checkbox element is split into two seperate  items, which on its own isn't a big deal because  they are next to each other in the order. However for NVDA the checkbox part isnt rendered at all, instead presenting a blank line. For narrator it is presented albeit as seperate item, but problem is it merges with the next item, like if the next thing is a heading, then the checkbox is presented as being part of the heading.

## Root Cause Analysis

`Element.Input.checkbox` is turned into this sort of html:
```
<label>
    label element...
    <div role="checkbox" aria-checked="true"></div>
</label>
```

testing around with the two screen readers I discovered that a `<div role="checkbox">`  element that doesnt contain any text as a descendant is presented by NVDA as a blank line. Even if the html element has a aria-label or is contained in a `<label>` such as here. 

I would say this is definitely a bug on part of the screen reader since if instead of using `<div role="checkbox">` one would use `<input type="checkbox" />` then the label is associated correctly and I will raise this issue with NVDA.

However I found by shuffling around the aria attributes that both NVDA and narrator can present the elm UI checkbox as if it was a native checkbox. So, since for the screen readers to update their behaviour to handle this way of representing a checkbox with aria attributes might take a long time and there seems to be a solution that only requires altering the aria, it is probably better to just implement the change in elm UI.

## solution

I tested a lot of different  html configurations, searching for one that minimized changes to html structure and worked uniformly for the different screen readers. Here is the best I found. Instead of putting the aria checkbox attributes on the `<div>`, just put it on the parent `<label>` like this:

```
<label role="checkbox" aria-checked="true" tabindex="0">
    label element...
    <div></div> <!-- checkbox icon -->
</label>
```

Testing with both NVDA and narrator, it now presents exactly as a native `<input type="checkbox" />`. And since the structure is the same and only placement of aria changes, hopefully this would have no impact on other things.

## Versions
 - OS: Windows 10 version 1903
 - Browser Chrome, Firefox
 - Browser Version
 - Elm Version 0.19.1
 - Elm UI Version 1.1.7

**Comment by SamKacer** [comment]
As an update, I've tested with VoiceOver on iOS 13.4 as well. There it runs into similar problems as narrator and the proposed solution works as well.

I also asked the community to test with other screen readers and so far I got feedback for Orca on linux, where it is same as narrator and solution also works.

[thread gathering community feedback](https://forum.audiogames.net/topic/35312/asking-for-jaws-and-talk-back-feedback-for-a-html-widget/)

Could you please create a branch with the proposed changes when you have time. Then I can do another round of testing to confirm whether it is more accessible.

**Comment by SamKacer** [comment]
Just got feedback for TalkBack on android in the forum thread linked above and the elm UI checkbox seems to have same problems as NVDA and the proposed solution works completely.

**Comment by SamKacer** [comment]
Just got feedback for JAWS. The elm UI checkbox does not appear correctly, simlarly as to narrator and the proposed solution presents as a classic checkbox. So, that's all the major screen readers on all major OS. Looks like all of them have problem with elm UI checkbox and for all of them the proposed solution is the same as a native checkbox.

**Comment by alexkorban** [comment]
#pain-point

**Comment by mdgriffith** [comment]
Hey Sam,

Sorry for the delay in replying here.  I really appreciate your breakdown and thoroughness of exploring the issue.

I've made the change directly on master and [published a small example form with a checkbox for you to verify](http://mdgriffith.github.io/elm-ui/accessible-checkbox.html).

If that form functions correctly I'll go ahead and release a patch version with these changes.

Thanks!

**Comment by SamKacer** [comment]
@mdgriffith 
Hi Matt,

happy to help. You could say I am invested in this sort of stuff :)

I just checked the example with NVDA, windows Narrator, and VoiceOver and for those it functions correctly now. I think it is probably safe to assume it is fixed for JAWS and TalkBack.

**Comment by mdgriffith** [comment]
Wonderful!  I'm putting together a release which should be out shortly.

**Comment by mdgriffith** [comment]
Released the fix as `1.1.8`!
